packages/material-ui/src/Typography/Typography.js

```diff
@@ -207,19 +294,29 @@ Typography.defaultProps = {
   color: 'default',
   gutterBottom: false,
   headlineMapping: {
+    h1: 'h1',
+    h2: 'h2',
+    h3: 'h3',
+    h4: 'h4',
+    h5: 'h5',
+    h6: 'h6',
+    subtitle1: 'h6',
+    subtitle2: 'h6',
+    body1: 'p',
+    body2: 'p',
+    // deprecated
     display4: 'h1',
     display3: 'h1',
     display2: 'h1',
     display1: 'h1',
     headline: 'h1',
     title: 'h2',
     subheading: 'h3',
-    body2: 'aside',
-    body1: 'p',
   },
   noWrap: false,
   paragraph: false,
+  suppressDeprecationWarnings: false,
```

Comment:
That's the "partial migration strategy" discussion all over again. We decided against that and now we reintroduce it by allowing partial suppression of deprecation warnings.

Reply:
The alternative of not introducing this property is to upgrade the variant usage in our core components. You have made a good point on raising it's a breaking change and should be deferred to v4.

Reply:
The alternative is to leave it as is and let the user decide how to treat the warning. We don't have to update internal usage now. But the user should know where he has to expect changes with the next major release. That is the hole point of deprecation warnings in a minor before a major release.

Updating variants in our internal components is scheduled for the next major release in a separate PR as.

Reply:
> But the user should know where he has to expect changes with the next major release.

What's the value in that?

Reply:
They can switch to the Typography v2 theme which we discussed is the preferred way. A global, immediate switch.